semgrep-core/src/engine/Match_rules.ml

```diff
@@ -796,48 +796,90 @@ and filter_ranges env xs cond =
          | R.CondGeneric e ->
              let env = Eval_generic.bindings_to_env bindings in
              Eval_generic.eval_bool env e
-         | R.CondPattern (mvar, formula) -> (
+         | R.CondPattern (mvar, opt_lang, formula) -> (
              (* If anything goes wrong the default is to filter out! *)
+             let lazy_ast_of_string lang str =
```

Comment:
I'm not a big fan of long nested functions. Maybe better to put that as an helper function out of filter_ranges.

Reply:
I agree, I was being a bit lazy here. I was actually considering to move the logic for `CondPattern` to separate function. But I'll just pull out these two helpers for now.

Reply:
I ended up taking `CondPattern`s logic into a separate function as well.